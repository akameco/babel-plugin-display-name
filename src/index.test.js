// @flow
import pluginTester from 'babel-plugin-tester'
import plugin from '.'

pluginTester({
  title: 'default',
  plugin,
  babelOptions: { parserOpts: { plugins: ['jsx', 'classProperties'] } },
  snapshot: true,
  tests: [
    {
      title: 'code',
      code: `
class Display extends React.Component {
  // @displayName
  static On = () => <div>On</div>
  static Off = () => <div>Off</div>
  render() {
    return <div>{this.props.children}</div>
  }
}
  `,
    },
  ],
})
