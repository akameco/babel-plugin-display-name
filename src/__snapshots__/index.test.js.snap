// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`default code: code 1`] = `
"
class Display extends React.Component {
  // @displayName
  static On = () => <div>On</div>
  static Off = () => <div>Off</div>
  render() {
    return <div>{this.props.children}</div>
  }
}

      ↓ ↓ ↓ ↓ ↓ ↓

class Display extends React.Component {
  // @displayName
  static On = () => <div>On</div>;
  static Off = () => <div>Off</div>;
  render() {
    return <div>{this.props.children}</div>;
  }
}
Display.On.displayName = \\"Display.On\\";
"
`;
